homeassistant/components/maxcube/climate.py

```diff
@@ -257,37 +211,25 @@ def preset_modes(self):
 
     def set_preset_mode(self, preset_mode):
         """Set new operation mode."""
-        device = self._cubehandle.cube.device_by_rf(self._rf_address)
-        temp = None
-        mode = MAX_DEVICE_MODE_AUTOMATIC
-
-        if preset_mode in [PRESET_COMFORT, PRESET_ECO, PRESET_ON]:
-            mode = MAX_DEVICE_MODE_MANUAL
-            if preset_mode == PRESET_COMFORT:
-                temp = device.comfort_temperature
-            elif preset_mode == PRESET_ECO:
-                temp = device.eco_temperature
-            else:
-                temp = ON_TEMPERATURE
-        else:
-            mode = HASS_PRESET_TO_MAX_MODE[preset_mode] or MAX_DEVICE_MODE_AUTOMATIC
-
-        with self._cubehandle.mutex:
-            try:
-                self._cubehandle.cube.set_temperature_mode(device, temp, mode)
-            except (socket.timeout, OSError):
-                _LOGGER.error("Setting operation mode failed")
-                return
+        if preset_mode == PRESET_COMFORT:
+            self._set_target(MAX_DEVICE_MODE_MANUAL, self._device.comfort_temperature)
+        elif preset_mode == PRESET_ECO:
+            self._set_target(MAX_DEVICE_MODE_MANUAL, self._device.eco_temperature)
+        elif preset_mode == PRESET_ON:
+            self._set_target(MAX_DEVICE_MODE_MANUAL, ON_TEMPERATURE)
+        elif preset_mode == PRESET_AWAY:
+            self._set_target(MAX_DEVICE_MODE_VACATION, None)
```

Comment:
`None` is not a float so either the typing needs to change on `self._set_target` or we need to call it differently.

Reply:
Both parameters of _set_target are optional. mode parameter defaults to current mode and temperature defaults to current temperature except when the mode is auto; in that case it read the temperature from the weekly schedule.

I will document the method properly.